services/sqs/src/main/java/software/amazon/awssdk/services/sqs/batchmanager/SqsAsyncBatchManager.java

```
@@ -0,0 +1,122 @@
+ * permissions and limitations under the License.
+ */
+
+package software.amazon.awssdk.services.sqs.batchmanager;
+
+import java.util.concurrent.CompletableFuture;
+import software.amazon.awssdk.annotations.SdkPublicApi;
+import software.amazon.awssdk.core.batchmanager.BatchOverrideConfiguration;
+import software.amazon.awssdk.services.sqs.SqsAsyncClient;
+import software.amazon.awssdk.services.sqs.internal.batchmanager.DefaultSqsAsyncBatchManager;
+import software.amazon.awssdk.services.sqs.model.ChangeMessageVisibilityRequest;
+import software.amazon.awssdk.services.sqs.model.ChangeMessageVisibilityResponse;
+import software.amazon.awssdk.services.sqs.model.DeleteMessageRequest;
+import software.amazon.awssdk.services.sqs.model.DeleteMessageResponse;
+import software.amazon.awssdk.services.sqs.model.SendMessageRequest;
+import software.amazon.awssdk.services.sqs.model.SendMessageResponse;
+import software.amazon.awssdk.utils.SdkAutoCloseable;
+
+/**
+ * Batch manager class that implements automatic batching features for a SQS async client. This can be created using the static
+ * {@link #builder()} method.
+ */
+@SdkPublicApi
+public interface SqsAsyncBatchManager extends SdkAutoCloseable {
+    /**
+     * Buffers outgoing SendMessageRequests on the client and sends them as a SendMessageBatchRequest to SQS. Requests are
+     * batched together according to a batchKey and are sent periodically to SQS. If the number of requests for a batchKey
+     * reaches or exceeds the configured max items, then the requests are immediately flushed and the timeout on the periodic
+     * flush is reset.
+     * <p>
+     * By default, messages are batched with a maximum batch size of 10. These settings can be customized via the configuration.
+     *
+     * @param message the outgoing SendMessageRequest.
+     * @return a CompletableFuture of the corresponding SendMessageResponse.
+     */
+    CompletableFuture<SendMessageResponse> sendMessage(SendMessageRequest message);
```

Should we throw UnsupportedOperation? That's what we do for the APIs in the low level client interface

---

Default behavior throws unsopportedOperation in SqsBatchManager and SqsAsyncBatchManager